File: Coverage/webapp/src/app/shared/dto/employes-flexibility.ts
import { Employe } from './employe';

export class EmployesFlexibility {

    flexible: Employe[];
    inflexible: Employe[];
    indifferent: Employe[];

    constructor(json: any) {
        this.flexible = json.flexible;
        this.inflexible = json.inflexible;
        this.indifferent = json.indifferent;
    }

    public getAll(): Employe[] {
        return this.inflexible.concat(this.flexible).concat(this.indifferent);
    }
}

File: Coverage/Dto/EmployesFlexibility.cs
using System.Collections.Generic;

namespace Seac.Coverage.Dto
{
    public class EmployesFlexibility
    {
        public IList<EmployeDto> Flexible { get; }
        public IList<EmployeDto> Inflexible { get; }
        public IList<EmployeDto> Indifferent { get; } = new List<EmployeDto>();

        public EmployesFlexibility(IList<EmployeDto> flexible, IList<EmployeDto> inflexible)
        {
            Flexible = flexible;
            Inflexible = inflexible;
        }

        public EmployesFlexibility(IList<EmployeDto> flexible, IList<EmployeDto> inflexible, IList<EmployeDto> indifferent) : this(flexible, inflexible)
        {
            Indifferent = indifferent;
        }
    }
}